cem/utils.c

```diff
@@ -48,26 +48,28 @@ void repeat_mem(void *dst, const int len, void *block, int block_len) {
   int offset;
 
   for (offset=0; offset < len; offset += block_len) {
-    memcpy(dst + offset, block, block_len);
+    memcpy((char *)dst + offset, block, block_len);
   }
-  memcpy(dst + len - last_block_len, block, last_block_len);
+  memcpy((char *)dst + len - last_block_len, block, last_block_len);
 }
 
 
 // Run vertical stripes down a CEM matrix, making certain to obey the
 // periodic boundary conditions.
 void stripe_cem_matrix(void **matrix, int n_rows, int n_cols, int itemsize,
-    void *stripe, int nitems)
+  void *stripe, int nitems)
 {
   const int offset_to_left = (n_cols / 4) * itemsize;
   const int width = (n_cols / 2) * itemsize;
   void *row = malloc(itemsize * n_cols);
   int i;
 
-  repeat_mem(row + offset_to_left, width, stripe, nitems * itemsize);
+  repeat_mem((char *)row + offset_to_left, width, stripe, nitems * itemsize);
   apply_periodic_boundary(row, itemsize, n_cols);
   for (i = 0; i < n_rows; i++)
-    memcpy(matrix[i], row, itemsize * n_cols);
+  {
+    memcpy((char *)matrix + (i * itemsize * n_cols), row, itemsize * n_cols);
```

Comment:
This change should fix the failing unit test (it does on my machine, anyway) and maybe (I hope) the error with `set_rock_blocks` you are seeing following your `TODO` comment in `cem.c`.